src/lightning/pytorch/trainer/connectors/accelerator_connector.py

```diff
@@ -496,8 +498,11 @@ def _check_and_init_precision(self) -> PrecisionPlugin:
         if isinstance(self._precision_plugin_flag, PrecisionPlugin):
             return self._precision_plugin_flag
 
-        if isinstance(self.accelerator, IPUAccelerator):
-            return IPUPrecisionPlugin(self._precision_flag)  # type: ignore
+        if _LIGHTNING_GRAPHCORE_AVAILABLE:
+            from lightning_graphcore import IPUAccelerator, IPUPrecision
+
+            if isinstance(self.accelerator, IPUAccelerator):
+                return IPUPrecision(self._precision_flag)
```

Comment:
(Not for this PR, just thinking out loud)

For the strategies that have a fixed precision class, we don't really need this logic in the accelerator. Since the strategy owns the precision plugin, the strategy.precision_plugin could be a no-op and then we wouldn't need this.

What do you think @awaelchli?

Reply:
probably yes